### ISSUE TITLE
Trollop dependency is deprecated

### ISSUE THREAD

**Issue by DannyBen** [thomasfl/filewatcher#92]
When installing filewatcher with bundler, trollop shows this message:

```
Post-install message from trollop:
!    The 'trollop' gem has been deprecated and has been replaced by 'optimist'.
!    See: https://rubygems.org/gems/optimist
!    And: https://github.com/ManageIQ/optimist
```

**Comment by AlexWayfer** [comment]
> Are you planning to make this change?

Yep, I'm planning to dig into it. Maybe we'll have more easier-to-test CLI library.

**Comment by DannyBen** [comment]
In my opinion, the behavior of trollop/optimist is a little aggressive. It is one thing to show a deprecation warning when installing with bundler, but they are also showing a deprecation warning whenever anything that uses this gem is executed:

```
$ filewatcher
[DEPRECATION] This gem has been renamed to optimist and will no longer be suppor
ted. Please switch to optimist as soon as possible.
Filewatcher scans the filesystem and executes shell commands when files
changes.

Usage:
   ...
```

This by itself is reason enough to abandon this optimist gem.

**Comment by AlexWayfer** [comment]
> It is one thing to show a deprecation warning when installing with bundler, but they are also showing a deprecation warning whenever anything that uses this gem is executed:

Few people look at the `bundler` output, many look at application output, so I'm not against this.

There is no "right way", unfortunately.

I'll compare by features, maintenance, etc., not by occasionally warnings.

**Comment by AlexWayfer** [comment]
I personally like [Thor](https://github.com/erikhuda/thor) for its API and abilities, but it looks unmaintained (number of open issues and PRs). 

I saw `subcommands` in the [Optimist](https://github.com/ManageIQ/optimist) README, so, I think it'll suit us.

Thank you for report.

The new release of Filewatcher coming soon (I hope). I'm planning use it instead of Guard and polish it even more.

**Comment by thomasfl** [comment]
Thank's for fixing this @AlexWayfer, and thank's for reporting this @DannyBen. 

To be honest I didn't know the meaning of the word trollop when I first started using it on filewatcher. I am glad the authors have decided to rename trollop to the more neutral name optimist.

**Comment by AlexWayfer** [comment]
> To be honest I didn't know the meaning of the word trollop when I first started using it on filewatcher. I am glad the authors have decided to rename trollop to the more neutral name optimist.

Oh, I didn't too, before your comment… :no_mouth:

**Comment by thomasfl** [comment]
The authors of trollop have probably done what have to be done to make people update to the new gem name. The library works perfectly well. :-)

**Comment by AlexWayfer** [comment]
`1.1.0` with this change published.
